fix: spacebar panning in vueNodes mode

Myestery opened the pull request:
## Summary
- Forward keydown events to litegraph's `processKey` when Vue nodes have focus (litegraph only binds keydown to canvas)
- Sync pointer state with litegraph canvas during link dragging for spacebar panning while dragging connections
- Add tests for keydown forwarding behavior

## Test plan
- [ ] Enable vueNodes mode and "Drag Navigation" (legacy) mode
- [ ] Select a Vue node, hold spacebar and drag - canvas should pan
- [ ] Drag a link connection, hold spacebar and drag - canvas should pan while link stays attached

Fixes #7806

┆Issue is synchronized with this [Notion page](https://www.notion.so/PR-7807-fix-spacebar-panning-in-vueNodes-mode-2d96d73d3650813987a1e1f17a6d5bdd) by [Unito](https://www.unito.io)

Myestery committed:
fix: spacebar panning in vueNodes mode

Forward keydown events to litegraph's processKey when Vue nodes have
focus. Litegraph only binds keydown to the canvas element, so spacebar
panning didn't work when Vue nodes were focused.

Also sync pointer state with litegraph canvas during link dragging to
enable spacebar panning while dragging connections.

Fixes #7806

DrJKL commented on src/renderer/extensions/vueNodes/composables/useNodePointerInteractions.ts:
```diff
@@ -6,12 +6,34 @@ import { useVueNodeLifecycle } from '@/composables/graph/useVueNodeLifecycle'
+    },
+    true
+  )
+}
```
I don't want to add more arbitrary event listeners in various composables if we can help it.
Could you use [this](https://vueuse.org/core/useMagicKeys/#usemagickeys)?

Myestery committed:
refactor: address review comments - use useMagicKeys and derive button state

- Refactored spacebar forwarding to use useMagicKeys from VueUse instead of raw event listeners
- Added filtering for editable elements (input, textarea, contentEditable)
- Changed panning check in useSlotLinkInteraction to derive button state from event.buttons instead of storing canvas.pointer.isDown statefully
- Updated tests to work with the new useMagicKeys approach using vi.hoisted pattern

DrJKL commented on src/renderer/extensions/vueNodes/composables/useNodePointerInteractions.ts:
```diff
@@ -10,29 +12,51 @@ import { useNodeDrag } from '@/renderer/extensions/vueNodes/layout/useNodeDrag'
+  if (tag === 'INPUT' || tag === 'TEXTAREA') return true
+  if (el instanceof HTMLElement && el.isContentEditable) return true
+  return false
+}
```
Could we consolidate the logic here with the logic in `keybindingService`?

DrJKL replied:
Or just use keybindingService somehow?

Myestery replied:
so I just checked and its a bit complicated since it will require keybindingService to handle keyup events and we now need to take away the magickeys and alter graphview too

DrJKL commented on src/renderer/extensions/vueNodes/composables/useNodePointerInteractions.ts:
```diff
@@ -10,29 +12,51 @@ import { useNodeDrag } from '@/renderer/extensions/vueNodes/layout/useNodeDrag'
+  spacebarForwardingInitialized = true
+
+  const { space } = useMagicKeys()
+  const activeElement = useActiveElement()
```
I don't think we need to (or should) have these instantiated conditionally. We can just include them in the composable directly.

Myestery committed:
refactor: simplify spacebar forwarding to module-level initialization

Myestery committed:
fix: Explicitly pass null to `isEditableElement` when `activeElement.value` is falsy.

Myestery committed:
refactor: use read_only as signal, let events bubble to litegraph

- Set canvas.read_only directly when spacebar pressed
- Skip stopPropagation when read_only is true in slot link handler
- Remove manual dragging_canvas and processMouseMove management
- Litegraph handles panning through its normal event handlers

Myestery committed:
fix: enable spacebar panning during slot link drag

Move spacebar detection to document-level listener in LGraphCanvas when
vueNodesMode is enabled. Implement direct panning in useSlotLinkInteraction
when spacebar is held during connection drag, bypassing litegraph event
handling for smoother panning while maintaining link position updates.

Fixes #7806

Myestery committed:
refactor: consolidate panning state in litegraph

Remove duplicate panning state management from useSlotLinkInteraction.
Instead of Vue manually manipulating canvas.ds.offset, sync pointer.isDown
with litegraph and let events bubble when in panning mode (read_only).

- Remove isPanningDuringLinkDrag and lastPanningMouse local state
- Remove manual offset manipulation in handlePointerMove
- Conditionally stopPropagation only when NOT in panning mode
- Set canvas.pointer.isDown so spacebar triggers litegraph panning

Myestery committed:
Merge remote-tracking branch 'origin/main' into fix/spacebar-panning-vue-nodes

Myestery committed:
fix: always remove document keydown listener in unbindEvents

Remove the LiteGraph.vueNodesMode check when removing the document-level
keydown listener in unbindEvents. If vueNodesMode was true during bindEvents
but changed before unbindEvents, the listener would be left dangling.
Now guarded only by existence of _key_callback.

Myestery committed:
fix: skip text-editable elements in processKey

Broaden the target guard in processKey to skip all text-editable surfaces
(input, textarea, contenteditable) before handling shortcuts. This prevents
space/Ctrl+A/C from being blocked when typing in prompt textareas or other
multiline fields in Vue nodes mode.

actions-user committed:
[automated] Apply ESLint and Prettier fixes

Myestery committed:
fix: add capture flag to removeEventListener calls and reuse target variable

- Add capture:true to removeEventListener calls that were added with capture
  to prevent event listener leaks when unbinding
- Reuse existing target variable in Delete/Backspace check instead of
  re-accessing e.target with @ts-expect-error
- Add isContentEditable check to Delete/Backspace guard for consistency

Myestery committed:
Merge remote-tracking branch 'origin/main' into fix/spacebar-panning-vue-nodes

Myestery committed:
format files

Commit:
[automated] Update test expectations

DrJKL committed:
Merge branch 'main' into fix/spacebar-panning-vue-nodes
